### PR TITLE
feat: Add commit history between releases

### PR DESCRIPTION
**Pull request by soheeeeP**
issue #691 

- Add commit history link between releases in release notes(`CHANGELOG_RELEASE.md`) by updating `scripts/extract-release-changelog.py`

### CODE REVIEW

**Comment by rapsealk** [comment]
Towncrier does not recognize the changelog since the tag `ci-cd` is not supported. You can choose one from `["breaking", "feature", "fix", "deprecation", "doc", "misc"]`.
You may check details here: https://github.com/lablup/backend.ai/blob/6636e3db145a9778cfed60e7b0ccd9578f0ff69b/pyproject.toml#L11-L39

File: changes/721.feature.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add commit history link between releases in release notes(`CHANGELOG_RELEASE.md`) by updating `scripts/extract-release-changelog.py`
```
Comment (kyujin-cho) [inline changes/721.feature.md]: Sinse changelog is for informing "what" has changed - not "how" it's changed -, it'll be okay to skip which script you updated.
```suggestion
Add commit history link between releases in release notes(`CHANGELOG_RELEASE.md`).
```

**Review by kyujin-cho — CHANGES_REQUESTED** [review]

File: changes/721.feature.md (hunk 1 above)
Comment (soheeeeP) [reply changes/721.feature.md]: Thanks. I've updated the changelog on the below commit! (dfe35e1)

File: scripts/extract-release-changelog.py [hunk 2]
```diff
@@ -14,6 +14,13 @@ def get_tag():
     revision = p.stdout.decode().strip()
     return revision
 
+def get_prev_tag():
+    rev = subprocess.run(['git', 'rev-list', '--tags', '--skip=0', '--max-count=1'], capture_output=True)
+    rev = rev.stdout.decode().strip()
```
Comment (fregataa) [inline scripts/extract-release-changelog.py]: since the first `rev` is subprocess object, how about rename it?
this is just a suggestion.

File: scripts/extract-release-changelog.py [hunk 3]
```diff
@@ -14,6 +14,13 @@ def get_tag():
...
+
+    tag = subprocess.run(['git', 'describe', '--abbrev=0', '--tags', f'{rev}'], capture_output=True)
+    tag = tag.stdout.decode().strip()
```
Comment (fregataa) [inline scripts/extract-release-changelog.py]: same here. The first `tag` is subprocess object.

File: scripts/extract-release-changelog.py [hunk 4]
```diff
@@ -23,8 +30,11 @@ def main():
     )
     args = parser.parse_args()
 
-    tag = get_tag()
-    url = f"https://github.com/lablup/backend.ai/blob/{tag}/CHANGELOG.md"
+    prev_tag, tag = get_prev_tag(), get_tag()
+    url = {
+        "commit_log": f"https://github.com/lablup/backend.ai/compare/{prev_tag}...{tag}",
+        "change_log": f"https://github.com/lablup/backend.ai/blob/{tag}/CHANGELOG.md"
+    }
```
Comment (fregataa) [inline scripts/extract-release-changelog.py]: just a question.
why did you change `url` into dictionary?
we can just make multiple `..._url` strings.

**Review by fregataa — COMMENTED** [review]
Thank you for your contribution!
I left some comments for review. But it does not mean that you need to change them all into my suggestion or examples I wrote.
If you answer them all, please re-request review.

File: scripts/extract-release-changelog.py (hunk 4 above)
Comment (soheeeeP) [reply scripts/extract-release-changelog.py]: I just thought that using a dictionary would be convenient to handle multiple urls. (in the case of adding new URLs related to logs).

**Review by fregataa — APPROVED** [review]
LGTM.
And you need to re-request review after you update.

**Review by kyujin-cho — APPROVED** [review]
LGTM!
